libdnf/dnf-repo.cpp

```diff
@@ -1464,6 +1466,28 @@ dnf_repo_check_internal(DnfRepo *repo,
         return FALSE;
     }
 
+    /* init newRepo */
+    auto newRepo = hy_repo_create(priv->repo->getId().c_str());
+    auto newRepoImpl = libdnf::repoGetImpl(newRepo);
+
+    // "additionalMetadata" are not part of the config. They are filled during runtime
+    // (eg. by plugins) and must be kept.
+    newRepoImpl->additionalMetadata = repoImpl->additionalMetadata;
+
+    hy_repo_free(priv->repo);
+    priv->repo = newRepo;
+    newRepoImpl->repomdFn = yum_repo->repomd;
+    for (auto *elem = yum_repo->paths; elem; elem = g_slist_next(elem)) {
+        if (elem->data) {
+            auto yumrepopath = static_cast<LrYumRepoPath *>(elem->data);
+            newRepoImpl->metadataPaths[yumrepopath->type] = yumrepopath->path;
+        }
+    }
+
+    /* ensure we reset the values from the keyfile */
+    if (!dnf_repo_set_keyfile_data(repo, TRUE, error))
+        return FALSE;
+
```

Comment:
I had to move this section up so that the repo impl is properly set up for `isInSync()`